// box-node-sdk/lib-esm/managers/shield-information-barriers.generated.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
const url_path_1 = __importDefault(require("../util/url-path"));
/**
 */
class ShieldInformationBarrierManager {
    /**
     * @param {BoxClient} client The Box API Client that is responsible for making calls to the API
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Get shield information barrier with specified ID
     *
     * Get shield information barrier based on provided ID.
     * @param {object} options Options for the request
     * @param {string} options.shield_information_barrier_id The ID of the shield information barrier.
     * @param {Function} [callback] Passed the result if successful, error otherwise
     * @returns {Promise<schemas.ShieldInformationBarrier>} A promise resolving to the result or rejecting with an error
     */
    getById(options, callback) {
        const { shield_information_barrier_id: shieldInformationBarrierId, ...queryParams } = options, apiPath = (0, url_path_1.default)('shield_information_barriers', shieldInformationBarrierId), params = {
            qs: queryParams,
        };
        return this.client.wrapWithDefaultHandler(this.client.get)(apiPath, params, callback);
    }
    /**
     * List shield information barriers
     *
     * Retrieves a list of shield information barrier objects
     * for the enterprise of JWT.
     * @param {object} [options] Options for the request
     * @param {string} [options.marker] Defines the position marker at which to begin returning results. This is used when paginating using marker-based pagination.
     * @param {number} [options.limit] The maximum number of items to return per page.
     * @param {Function} [callback] Passed the result if successful, error otherwise
     * @returns {Promise<schemas.ShieldInformationBarriers>} A promise resolving to the result or rejecting with an error
     */
    getAll(options, callback) {
        const { ...queryParams } = options, apiPath = (0, url_path_1.default)('shield_information_barriers'), params = {
            qs: queryParams,
        };
        return this.client.wrapWithDefaultHandler(this.client.get)(apiPath, params, callback);
    }
    /**
     * Create shield information barrier
     *
     * Creates a shield information barrier to
     * separate individuals/groups within the same
     * firm and prevents confidential information passing between them.
     * @param {object} body
     * @param {object} [options] Options for the request
     * @param {Function} [callback] Passed the result if successful, error otherwise
     * @returns {Promise<schemas.ShieldInformationBarrier>} A promise resolving to the result or rejecting with an error
     */
    create(body, options, callback) {
        const { ...queryParams } = options, apiPath = (0, url_path_1.default)('shield_information_barriers'), params = {
            qs: queryParams,
            body: body,
        };
        return this.client.wrapWithDefaultHandler(this.client.post)(apiPath, params, callback);
    }
    /**
     * Add changed status of shield information barrier with specified ID
     *
     * Change status of shield information barrier with the specified ID.
     * @param {object} body
     * @param {object} [options] Options for the request
     * @param {Function} [callback] Passed the result if successful, error otherwise
     * @returns {Promise<schemas.ShieldInformationBarrier>} A promise resolving to the result or rejecting with an error
     */
    changeStatusById(body, options, callback) {
        const { ...queryParams } = options, apiPath = (0, url_path_1.default)('shield_information_barriers', 'change_status'), params = {
            qs: queryParams,
            body: body,
        };
        return this.client.wrapWithDefaultHandler(this.client.post)(apiPath, params, callback);
    }
}
module.exports = ShieldInformationBarrierManager;
