// box-typescript-sdk-gen/lib-esm/test/commons.generated.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getCcgAuth = getCcgAuth;
exports.getJwtAuth = getJwtAuth;
exports.getDefaultClientWithUserSubject = getDefaultClientWithUserSubject;
exports.getDefaultClient = getDefaultClient;
exports.createNewFolder = createNewFolder;
exports.uploadNewFile = uploadNewFile;
exports.getOrCreateTermsOfServices = getOrCreateTermsOfServices;
exports.getOrCreateClassification = getOrCreateClassification;
exports.getOrCreateClassificationTemplate = getOrCreateClassificationTemplate;
exports.getOrCreateShieldInformationBarrier = getOrCreateShieldInformationBarrier;
const classifications_generated_js_1 = require("../managers/classifications.generated.js");
const classifications_generated_js_2 = require("../managers/classifications.generated.js");
const utils_js_1 = require("../internal/utils.js");
const utils_js_2 = require("../internal/utils.js");
const utils_js_3 = require("../internal/utils.js");
const utils_js_4 = require("../internal/utils.js");
const client_generated_js_1 = require("../client.generated.js");
const classificationTemplate_generated_js_1 = require("../schemas/classificationTemplate.generated.js");
const ccgAuth_generated_js_1 = require("../box/ccgAuth.generated.js");
const ccgAuth_generated_js_2 = require("../box/ccgAuth.generated.js");
const utils_js_5 = require("../internal/utils.js");
const jwtAuth_generated_js_1 = require("../box/jwtAuth.generated.js");
const jwtAuth_generated_js_2 = require("../box/jwtAuth.generated.js");
const utils_js_6 = require("../internal/utils.js");
function getCcgAuth() {
    const ccgConfig = new ccgAuth_generated_js_2.CcgConfig({
        clientId: (0, utils_js_2.getEnvVar)('CLIENT_ID'),
        clientSecret: (0, utils_js_2.getEnvVar)('CLIENT_SECRET'),
        enterpriseId: (0, utils_js_2.getEnvVar)('ENTERPRISE_ID'),
    });
    const auth = new ccgAuth_generated_js_1.BoxCcgAuth({ config: ccgConfig });
    return auth;
}
function getJwtAuth() {
    const jwtConfig = jwtAuth_generated_js_2.JwtConfig.fromConfigJsonString((0, utils_js_1.decodeBase64)((0, utils_js_2.getEnvVar)('JWT_CONFIG_BASE_64')));
    const auth = new jwtAuth_generated_js_1.BoxJwtAuth({ config: jwtConfig });
    return auth;
}
function getDefaultClientWithUserSubject(userId) {
    if ((0, utils_js_5.isBrowser)()) {
        const ccgAuth = getCcgAuth();
        const ccgAuthUser = ccgAuth.withUserSubject(userId);
        return new client_generated_js_1.BoxClient({ auth: ccgAuthUser });
    }
    const auth = getJwtAuth();
    const authUser = auth.withUserSubject(userId);
    return new client_generated_js_1.BoxClient({ auth: authUser });
}
function getDefaultClient() {
    const client = new client_generated_js_1.BoxClient({
        auth: (0, utils_js_5.isBrowser)() ? getCcgAuth() : getJwtAuth(),
    });
    return client;
}
async function createNewFolder() {
    const client = getDefaultClient();
    const newFolderName = (0, utils_js_3.getUuid)();
    return await client.folders.createFolder({
        name: newFolderName,
        parent: { id: '0' },
    });
}
async function uploadNewFile() {
    const client = getDefaultClient();
    const newFileName = ''.concat((0, utils_js_3.getUuid)(), '.pdf');
    const fileContentStream = (0, utils_js_4.generateByteStream)(1024 * 1024);
    const uploadedFiles = await client.uploads.uploadFile({
        attributes: {
            name: newFileName,
            parent: { id: '0' },
        },
        file: fileContentStream,
    });
    return uploadedFiles.entries[0];
}
async function getOrCreateTermsOfServices() {
    const client = getDefaultClient();
    const tos = await client.termsOfServices.getTermsOfService();
    const numberOfTos = tos.entries.length;
    if (numberOfTos >= 1) {
        const firstTos = tos.entries[0];
        if ((0, utils_js_6.toString)(firstTos.tosType) == 'managed') {
            return firstTos;
        }
    }
    if (numberOfTos >= 2) {
        const secondTos = tos.entries[1];
        if ((0, utils_js_6.toString)(secondTos.tosType) == 'managed') {
            return secondTos;
        }
    }
    return await client.termsOfServices.createTermsOfService({
        status: 'disabled',
        tosType: 'managed',
        text: 'Test TOS',
    });
}
async function getOrCreateClassification(classificationTemplateInput) {
    const classificationTemplate = new classificationTemplate_generated_js_1.ClassificationTemplate({
        id: classificationTemplateInput.id,
        type: classificationTemplateInput.type,
        scope: classificationTemplateInput.scope,
        templateKey: classificationTemplateInput.templateKey,
        displayName: classificationTemplateInput.displayName,
        hidden: classificationTemplateInput.hidden,
        copyInstanceOnItemCopy: classificationTemplateInput.copyInstanceOnItemCopy,
        fields: classificationTemplateInput.fields,
    });
    const client = getDefaultClient();
    const classifications = classificationTemplate.fields[0].options;
    const currentNumberOfClassifications = classifications.length;
    if (currentNumberOfClassifications == 0) {
        const classificationTemplateWithNewClassification = await client.classifications.addClassification([
            new classifications_generated_js_1.AddClassificationRequestBody({
                data: {
                    key: (0, utils_js_3.getUuid)(),
                    staticConfig: {
                        classification: {
                            colorId: 3,
                            classificationDefinition: 'Some description',
                        },
                    },
                },
            }),
        ]);
        return classificationTemplateWithNewClassification.fields[0].options[0];
    }
    return classifications[0];
}
async function getOrCreateClassificationTemplate() {
    const client = getDefaultClient();
    try {
        return await client.classifications.getClassificationTemplate();
    }
    catch (error) {
        return await client.classifications.createClassificationTemplate({
            fields: [
                new classifications_generated_js_2.CreateClassificationTemplateRequestBodyFieldsField({ options: [] }),
            ],
        });
    }
    finally {
    }
}
async function getOrCreateShieldInformationBarrier(clientInput, enterpriseId) {
    const client = new client_generated_js_1.BoxClient({
        auth: clientInput.auth,
        networkSession: clientInput.networkSession,
    });
    const barriers = await client.shieldInformationBarriers.getShieldInformationBarriers();
    const numberOfBarriers = barriers.entries.length;
    if (numberOfBarriers == 0) {
        return await client.shieldInformationBarriers.createShieldInformationBarrier({
            enterprise: { id: enterpriseId },
        });
    }
    return barriers.entries[numberOfBarriers - 1];
}
