// box-typescript-sdk-gen/lib-esm/test/comments.generated.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.client = void 0;
const utils_js_1 = require("../internal/utils.js");
const utils_js_2 = require("../internal/utils.js");
const commons_generated_js_1 = require("./commons.generated.js");
exports.client = (0, commons_generated_js_1.getDefaultClient)();
test('testComments', async function testComments() {
    const fileSize = 256;
    const fileName = (0, utils_js_2.getUuid)();
    const fileByteStream = (0, utils_js_1.generateByteStream)(fileSize);
    const parentId = '0';
    const uploadedFiles = await exports.client.uploads.uploadFile({
        attributes: {
            name: fileName,
            parent: {
                id: parentId,
            },
        },
        file: fileByteStream,
    });
    const fileId = uploadedFiles.entries[0].id;
    const comments = await exports.client.comments.getFileComments(fileId);
    if (!(comments.totalCount == 0)) {
        throw new Error('Assertion failed');
    }
    const message = 'Hello there!';
    const newComment = await exports.client.comments.createComment({
        message: message,
        item: {
            id: fileId,
            type: 'file',
        },
    });
    if (!(newComment.message == message)) {
        throw new Error('Assertion failed');
    }
    if (!(newComment.isReplyComment == false)) {
        throw new Error('Assertion failed');
    }
    if (!(newComment.item.id == fileId)) {
        throw new Error('Assertion failed');
    }
    const newReplyComment = await exports.client.comments.createComment({
        message: message,
        item: {
            id: newComment.id,
            type: 'comment',
        },
    });
    if (!(newReplyComment.message == message)) {
        throw new Error('Assertion failed');
    }
    if (!(newReplyComment.isReplyComment == true)) {
        throw new Error('Assertion failed');
    }
    const newMessage = 'Hi!';
    await exports.client.comments.updateCommentById(newReplyComment.id, {
        requestBody: { message: newMessage },
    });
    const newComments = await exports.client.comments.getFileComments(fileId);
    if (!(newComments.totalCount == 2)) {
        throw new Error('Assertion failed');
    }
    if (!(newComments.entries[1].message == newMessage)) {
        throw new Error('Assertion failed');
    }
    const receivedComment = await exports.client.comments.getCommentById(newComment.id);
    if (!(receivedComment.message == newComment.message)) {
        throw new Error('Assertion failed');
    }
    await exports.client.comments.deleteCommentById(newComment.id);
    await expect(async () => {
        await exports.client.comments.getCommentById(newComment.id);
    }).rejects.toThrow();
    await exports.client.files.deleteFileById(fileId);
});
