// box-typescript-sdk-gen/lib-esm/test/fileWatermarks.generated.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.client = void 0;
const fileWatermarks_generated_js_1 = require("../managers/fileWatermarks.generated.js");
const utils_js_1 = require("../internal/utils.js");
const utils_js_2 = require("../internal/utils.js");
const commons_generated_js_1 = require("./commons.generated.js");
exports.client = (0, commons_generated_js_1.getDefaultClient)();
test('testCreateGetDeleteFileWatermark', async function testCreateGetDeleteFileWatermark() {
    const fileName = ''.concat((0, utils_js_1.getUuid)(), '.txt');
    const uploadedFiles = await exports.client.uploads.uploadFile({
        attributes: {
            name: fileName,
            parent: { id: '0' },
        },
        file: (0, utils_js_2.generateByteStream)(10),
    });
    const file = uploadedFiles.entries[0];
    const createdWatermark = await exports.client.fileWatermarks.updateFileWatermark(file.id, {
        watermark: new fileWatermarks_generated_js_1.UpdateFileWatermarkRequestBodyWatermarkField({
            imprint: 'default',
        }),
    });
    const watermark = await exports.client.fileWatermarks.getFileWatermark(file.id);
    await exports.client.fileWatermarks.deleteFileWatermark(file.id);
    await expect(async () => {
        await exports.client.fileWatermarks.getFileWatermark(file.id);
    }).rejects.toThrow();
    await exports.client.files.deleteFileById(file.id);
});
