// box-typescript-sdk-gen/lib-esm/test/fileRequests.generated.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const utils_js_1 = require("../internal/utils.js");
const commons_generated_js_1 = require("./commons.generated.js");
const utils_js_2 = require("../internal/utils.js");
test('testGetCopyUpdateDeleteFileRequest', async function testGetCopyUpdateDeleteFileRequest() {
    const fileRequestId = (0, utils_js_1.getEnvVar)('BOX_FILE_REQUEST_ID');
    const userId = (0, utils_js_1.getEnvVar)('USER_ID');
    const client = (0, commons_generated_js_1.getDefaultClientWithUserSubject)(userId);
    const fileRequest = await client.fileRequests.getFileRequestById(fileRequestId);
    if (!(fileRequest.id == fileRequestId)) {
        throw new Error('Assertion failed');
    }
    if (!((0, utils_js_2.toString)(fileRequest.type) == 'file_request')) {
        throw new Error('Assertion failed');
    }
    const copiedFileRequest = await client.fileRequests.createFileRequestCopy(fileRequestId, {
        folder: {
            id: fileRequest.folder.id,
            type: 'folder',
        },
    });
    if (!!(copiedFileRequest.id == fileRequestId)) {
        throw new Error('Assertion failed');
    }
    if (!(copiedFileRequest.title == fileRequest.title)) {
        throw new Error('Assertion failed');
    }
    if (!(copiedFileRequest.description == fileRequest.description)) {
        throw new Error('Assertion failed');
    }
    const updatedFileRequest = await client.fileRequests.updateFileRequestById(copiedFileRequest.id, {
        title: 'updated title',
        description: 'updated description',
    });
    if (!(updatedFileRequest.id == copiedFileRequest.id)) {
        throw new Error('Assertion failed');
    }
    if (!(updatedFileRequest.title == 'updated title')) {
        throw new Error('Assertion failed');
    }
    if (!(updatedFileRequest.description == 'updated description')) {
        throw new Error('Assertion failed');
    }
    await client.fileRequests.deleteFileRequestById(updatedFileRequest.id);
    await expect(async () => {
        await client.fileRequests.getFileRequestById(updatedFileRequest.id);
    }).rejects.toThrow();
});
