// box-typescript-sdk-gen/lib-esm/test/folderWatermarks.generated.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.client = void 0;
const folderWatermarks_generated_js_1 = require("../managers/folderWatermarks.generated.js");
const utils_js_1 = require("../internal/utils.js");
const commons_generated_js_1 = require("./commons.generated.js");
exports.client = (0, commons_generated_js_1.getDefaultClient)();
test('testCreateGetDeleteFolderWatermark', async function testCreateGetDeleteFolderWatermark() {
    const folderName = (0, utils_js_1.getUuid)();
    const folder = await exports.client.folders.createFolder({
        name: folderName,
        parent: { id: '0' },
    });
    const createdWatermark = await exports.client.folderWatermarks.updateFolderWatermark(folder.id, {
        watermark: new folderWatermarks_generated_js_1.UpdateFolderWatermarkRequestBodyWatermarkField({
            imprint: 'default',
        }),
    });
    const watermark = await exports.client.folderWatermarks.getFolderWatermark(folder.id);
    await exports.client.folderWatermarks.deleteFolderWatermark(folder.id);
    await expect(async () => {
        await exports.client.folderWatermarks.getFolderWatermark(folder.id);
    }).rejects.toThrow();
    await exports.client.folders.deleteFolderById(folder.id);
});
