// box-typescript-sdk-gen/lib-esm/test/groups.generated.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.client = void 0;
const utils_js_1 = require("../internal/utils.js");
const commons_generated_js_1 = require("./commons.generated.js");
exports.client = (0, commons_generated_js_1.getDefaultClient)();
test('test_get_groups', async function test_get_groups() {
    const groups = await exports.client.groups.getGroups();
    if (!(groups.totalCount >= 0)) {
        throw new Error('Assertion failed');
    }
});
test('test_create_get_delete_group', async function test_create_get_delete_group() {
    const groupName = (0, utils_js_1.getUuid)();
    const groupDescription = 'Group description';
    const group = await exports.client.groups.createGroup({
        name: groupName,
        description: groupDescription,
    });
    if (!(group.name == groupName)) {
        throw new Error('Assertion failed');
    }
    const groupById = await exports.client.groups.getGroupById(group.id, {
        queryParams: {
            fields: ['id', 'name', 'description', 'group_type'],
        },
    });
    if (!(groupById.id == group.id)) {
        throw new Error('Assertion failed');
    }
    if (!(groupById.description == groupDescription)) {
        throw new Error('Assertion failed');
    }
    const updatedGroupName = (0, utils_js_1.getUuid)();
    const updatedGroup = await exports.client.groups.updateGroupById(group.id, {
        requestBody: {
            name: updatedGroupName,
        },
    });
    if (!(updatedGroup.name == updatedGroupName)) {
        throw new Error('Assertion failed');
    }
    await exports.client.groups.deleteGroupById(group.id);
    await expect(async () => {
        await exports.client.groups.getGroupById(group.id);
    }).rejects.toThrow();
});
