// box-typescript-sdk-gen/lib-esm/test/invites.generated.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const utils_js_1 = require("../internal/utils.js");
const commons_generated_js_1 = require("./commons.generated.js");
const utils_js_2 = require("../internal/utils.js");
test('testInvites', async function testInvites() {
    const userId = (0, utils_js_1.getEnvVar)('USER_ID');
    const client = (0, commons_generated_js_1.getDefaultClientWithUserSubject)(userId);
    const currentUser = await client.users.getUserMe({
        fields: ['enterprise'],
    });
    const email = (0, utils_js_1.getEnvVar)('BOX_EXTERNAL_USER_EMAIL');
    const invitation = await client.invites.createInvite({
        enterprise: {
            id: currentUser.enterprise.id,
        },
        actionableBy: {
            login: email,
        },
    });
    if (!((0, utils_js_2.toString)(invitation.type) == 'invite')) {
        throw new Error('Assertion failed');
    }
    if (!(invitation.invitedTo.id == currentUser.enterprise.id)) {
        throw new Error('Assertion failed');
    }
    if (!(invitation.actionableBy.login == email)) {
        throw new Error('Assertion failed');
    }
    const getInvitation = await client.invites.getInviteById(invitation.id);
    if (!(getInvitation.id == invitation.id)) {
        throw new Error('Assertion failed');
    }
});
