// box-typescript-sdk-gen/lib-esm/test/legalHoldPolicies.generated.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.client = void 0;
const utils_js_1 = require("../internal/utils.js");
const utils_js_2 = require("../internal/utils.js");
const utils_js_3 = require("../internal/utils.js");
const commons_generated_js_1 = require("./commons.generated.js");
exports.client = (0, commons_generated_js_1.getDefaultClient)();
test('testCreateNotOngoingLegalHoldPolicy', async function testCreateNotOngoingLegalHoldPolicy() {
    const legalHoldPolicyName = (0, utils_js_1.getUuid)();
    const legalHoldDescription = 'test description';
    const filterStartedAt = (0, utils_js_2.dateTimeFromString)('2021-01-01T00:00:00-08:00');
    const filterEndedAt = (0, utils_js_2.dateTimeFromString)('2022-01-01T00:00:00-08:00');
    const legalHoldPolicy = await exports.client.legalHoldPolicies.createLegalHoldPolicy({
        policyName: legalHoldPolicyName,
        description: legalHoldDescription,
        isOngoing: false,
        filterStartedAt: filterStartedAt,
        filterEndedAt: filterEndedAt,
    });
    if (!(legalHoldPolicy.policyName == legalHoldPolicyName)) {
        throw new Error('Assertion failed');
    }
    if (!(legalHoldPolicy.description == legalHoldDescription)) {
        throw new Error('Assertion failed');
    }
    if (!((0, utils_js_3.dateTimeToString)(legalHoldPolicy.filterStartedAt) ==
        (0, utils_js_3.dateTimeToString)(filterStartedAt))) {
        throw new Error('Assertion failed');
    }
    if (!((0, utils_js_3.dateTimeToString)(legalHoldPolicy.filterEndedAt) ==
        (0, utils_js_3.dateTimeToString)(filterEndedAt))) {
        throw new Error('Assertion failed');
    }
    await exports.client.legalHoldPolicies.deleteLegalHoldPolicyById(legalHoldPolicy.id);
});
test('testCreateUpdateGetDeleteLegalHoldPolicy', async function testCreateUpdateGetDeleteLegalHoldPolicy() {
    const legalHoldPolicyName = (0, utils_js_1.getUuid)();
    const legalHoldDescription = 'test description';
    const legalHoldPolicy = await exports.client.legalHoldPolicies.createLegalHoldPolicy({
        policyName: legalHoldPolicyName,
        description: legalHoldDescription,
        isOngoing: true,
    });
    if (!(legalHoldPolicy.policyName == legalHoldPolicyName)) {
        throw new Error('Assertion failed');
    }
    if (!(legalHoldPolicy.description == legalHoldDescription)) {
        throw new Error('Assertion failed');
    }
    const legalHoldPolicyId = legalHoldPolicy.id;
    const legalHoldPolicyById = await exports.client.legalHoldPolicies.getLegalHoldPolicyById(legalHoldPolicyId);
    if (!(legalHoldPolicyById.id == legalHoldPolicyId)) {
        throw new Error('Assertion failed');
    }
    const legalHoldPolicies = await exports.client.legalHoldPolicies.getLegalHoldPolicies();
    if (!(legalHoldPolicies.entries.length > 0)) {
        throw new Error('Assertion failed');
    }
    const updatedLegalHoldPolicyName = (0, utils_js_1.getUuid)();
    const updatedLegalHoldPolicy = await exports.client.legalHoldPolicies.updateLegalHoldPolicyById(legalHoldPolicyId, {
        requestBody: {
            policyName: updatedLegalHoldPolicyName,
        },
    });
    if (!(updatedLegalHoldPolicy.policyName == updatedLegalHoldPolicyName)) {
        throw new Error('Assertion failed');
    }
    await exports.client.legalHoldPolicies.deleteLegalHoldPolicyById(legalHoldPolicyId);
});
