// box-typescript-sdk-gen/lib-esm/test/legalHoldPolicyAssignments.generated.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.client = void 0;
const utils_js_1 = require("../internal/utils.js");
const commons_generated_js_1 = require("./commons.generated.js");
const commons_generated_js_2 = require("./commons.generated.js");
const utils_js_2 = require("../internal/utils.js");
exports.client = (0, commons_generated_js_1.getDefaultClient)();
test('testLegalHoldPolicyAssignments', async function testLegalHoldPolicyAssignments() {
    const legalHoldPolicyName = (0, utils_js_1.getUuid)();
    const legalHoldDescription = 'test description';
    const legalHoldPolicy = await exports.client.legalHoldPolicies.createLegalHoldPolicy({
        policyName: legalHoldPolicyName,
        description: legalHoldDescription,
        isOngoing: true,
    });
    const legalHoldPolicyId = legalHoldPolicy.id;
    const file = await (0, commons_generated_js_2.uploadNewFile)();
    const fileId = file.id;
    const legalHoldPolicyAssignment = await exports.client.legalHoldPolicyAssignments.createLegalHoldPolicyAssignment({
        policyId: legalHoldPolicyId,
        assignTo: {
            type: 'file',
            id: fileId,
        },
    });
    if (!((0, utils_js_2.toString)(legalHoldPolicyAssignment.legalHoldPolicy.type) ==
        'legal_hold_policy')) {
        throw new Error('Assertion failed');
    }
    if (!(legalHoldPolicyAssignment.assignedTo.id == fileId)) {
        throw new Error('Assertion failed');
    }
    if (!((0, utils_js_2.toString)(legalHoldPolicyAssignment.assignedTo.type) == 'file')) {
        throw new Error('Assertion failed');
    }
    const legalHoldPolicyAssignmentId = legalHoldPolicyAssignment.id;
    const legalHoldPolicyAssignmentFromApi = await exports.client.legalHoldPolicyAssignments.getLegalHoldPolicyAssignmentById(legalHoldPolicyAssignmentId);
    if (!(legalHoldPolicyAssignmentFromApi.id == legalHoldPolicyAssignmentId)) {
        throw new Error('Assertion failed');
    }
    const legalPolicyAssignments = await exports.client.legalHoldPolicyAssignments.getLegalHoldPolicyAssignments({
        policyId: legalHoldPolicyId,
    });
    if (!(legalPolicyAssignments.entries.length == 1)) {
        throw new Error('Assertion failed');
    }
    const filesOnHold = await exports.client.legalHoldPolicyAssignments.getLegalHoldPolicyAssignmentFileOnHold(legalHoldPolicyAssignmentId);
    if (!(filesOnHold.entries.length == 1)) {
        throw new Error('Assertion failed');
    }
    if (!(filesOnHold.entries[0].id == fileId)) {
        throw new Error('Assertion failed');
    }
    await exports.client.legalHoldPolicyAssignments.deleteLegalHoldPolicyAssignmentById(legalHoldPolicyAssignmentId);
    await expect(async () => {
        await exports.client.legalHoldPolicyAssignments.deleteLegalHoldPolicyAssignmentById(legalHoldPolicyAssignmentId);
    }).rejects.toThrow();
    await exports.client.files.deleteFileById(fileId);
    try {
        await exports.client.legalHoldPolicies.deleteLegalHoldPolicyById(legalHoldPolicyId);
    }
    catch (error) {
        console.log(''.concat('Could not delete Legal Policy with id: ', legalHoldPolicyId));
    }
    finally {
    }
});
