// box-typescript-sdk-gen/lib-esm/test/listCollaborations.generated.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const utils_js_1 = require("../internal/utils.js");
const utils_js_2 = require("../internal/utils.js");
const commons_generated_js_1 = require("./commons.generated.js");
const commons_generated_js_2 = require("./commons.generated.js");
const commons_generated_js_3 = require("./commons.generated.js");
const utils_js_3 = require("../internal/utils.js");
test('testListCollaborations', async function testListCollaborations() {
    const client = (0, commons_generated_js_1.getDefaultClient)();
    const folder = await (0, commons_generated_js_2.createNewFolder)();
    const file = await (0, commons_generated_js_3.uploadNewFile)();
    const group = await client.groups.createGroup({
        name: (0, utils_js_1.getUuid)(),
    });
    const groupCollaboration = await client.userCollaborations.createCollaboration({
        item: {
            type: 'folder',
            id: folder.id,
        },
        accessibleBy: {
            type: 'group',
            id: group.id,
        },
        role: 'editor',
    });
    const fileCollaboration = await client.userCollaborations.createCollaboration({
        item: {
            type: 'file',
            id: file.id,
        },
        accessibleBy: {
            type: 'user',
            id: (0, utils_js_2.getEnvVar)('USER_ID'),
        },
        role: 'editor',
    });
    if (!((0, utils_js_3.toString)(groupCollaboration.role) == 'editor')) {
        throw new Error('Assertion failed');
    }
    if (!((0, utils_js_3.toString)(groupCollaboration.type) == 'collaboration')) {
        throw new Error('Assertion failed');
    }
    const fileCollaborations = await client.listCollaborations.getFileCollaborations(file.id);
    if (!(fileCollaborations.entries.length > 0)) {
        throw new Error('Assertion failed');
    }
    const folderCollaborations = await client.listCollaborations.getFolderCollaborations(folder.id);
    if (!(folderCollaborations.entries.length > 0)) {
        throw new Error('Assertion failed');
    }
    const pendingCollaborations = await client.listCollaborations.getCollaborations({
        status: 'pending',
    });
    if (!(pendingCollaborations.entries.length >= 0)) {
        throw new Error('Assertion failed');
    }
    const groupCollaborations = await client.listCollaborations.getGroupCollaborations(group.id);
    if (!(groupCollaborations.entries.length > 0)) {
        throw new Error('Assertion failed');
    }
    await client.userCollaborations.deleteCollaborationById(groupCollaboration.id);
    await client.files.deleteFileById(file.id);
    await client.folders.deleteFolderById(folder.id);
    await client.groups.deleteGroupById(group.id);
});
