// box-typescript-sdk-gen/lib-esm/test/retentionPolicies.generated.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.client = void 0;
const utils_js_1 = require("../internal/utils.js");
const commons_generated_js_1 = require("./commons.generated.js");
exports.client = (0, commons_generated_js_1.getDefaultClient)();
test('testCreateUpdateGetDeleteRetentionPolicy', async function testCreateUpdateGetDeleteRetentionPolicy() {
    const retentionPolicyName = (0, utils_js_1.getUuid)();
    const retentionDescription = 'test description';
    const retentionPolicy = await exports.client.retentionPolicies.createRetentionPolicy({
        policyName: retentionPolicyName,
        policyType: 'finite',
        areOwnersNotified: true,
        canOwnerExtendRetention: true,
        description: retentionDescription,
        dispositionAction: 'remove_retention',
        retentionLength: '1',
        retentionType: 'modifiable',
    });
    if (!(retentionPolicy.policyName == retentionPolicyName)) {
        throw new Error('Assertion failed');
    }
    if (!(retentionPolicy.description == retentionDescription)) {
        throw new Error('Assertion failed');
    }
    const retentionPolicyById = await exports.client.retentionPolicies.getRetentionPolicyById(retentionPolicy.id);
    if (!(retentionPolicyById.id == retentionPolicy.id)) {
        throw new Error('Assertion failed');
    }
    const retentionPolicies = await exports.client.retentionPolicies.getRetentionPolicies();
    if (!(retentionPolicies.entries.length > 0)) {
        throw new Error('Assertion failed');
    }
    const updatedRetentionPolicyName = (0, utils_js_1.getUuid)();
    const updatedRetentionPolicy = await exports.client.retentionPolicies.updateRetentionPolicyById(retentionPolicy.id, {
        requestBody: {
            policyName: updatedRetentionPolicyName,
        },
    });
    if (!(updatedRetentionPolicy.policyName == updatedRetentionPolicyName)) {
        throw new Error('Assertion failed');
    }
    await exports.client.retentionPolicies.deleteRetentionPolicyById(retentionPolicy.id);
});
