// box-typescript-sdk-gen/lib-esm/test/retentionPolicyAssignments.generated.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.client = void 0;
const utils_js_1 = require("../internal/utils.js");
const utils_js_2 = require("../internal/utils.js");
const commons_generated_js_1 = require("./commons.generated.js");
exports.client = (0, commons_generated_js_1.getDefaultClient)();
test('testCreateUpdateGetDeleteRetentionPolicyAssignment', async function testCreateUpdateGetDeleteRetentionPolicyAssignment() {
    const retentionPolicyName = (0, utils_js_1.getUuid)();
    const retentionDescription = 'test description';
    const retentionPolicy = await exports.client.retentionPolicies.createRetentionPolicy({
        policyName: retentionPolicyName,
        policyType: 'finite',
        areOwnersNotified: true,
        canOwnerExtendRetention: true,
        description: retentionDescription,
        dispositionAction: 'remove_retention',
        retentionLength: '1',
        retentionType: 'modifiable',
    });
    const folder = await exports.client.folders.createFolder({
        name: (0, utils_js_1.getUuid)(),
        parent: { id: '0' },
    });
    const files = await exports.client.uploads.uploadFile({
        attributes: {
            name: (0, utils_js_1.getUuid)(),
            parent: {
                id: folder.id,
            },
        },
        file: (0, utils_js_2.generateByteStream)(10),
    });
    const file = files.entries[0];
    const newVersions = await exports.client.uploads.uploadFileVersion(file.id, {
        attributes: {
            name: (0, utils_js_1.getUuid)(),
        },
        file: (0, utils_js_2.generateByteStream)(20),
    });
    const newVersion = newVersions.entries[0];
    const retentionPolicyAssignment = await exports.client.retentionPolicyAssignments.createRetentionPolicyAssignment({
        policyId: retentionPolicy.id,
        assignTo: {
            type: 'folder',
            id: folder.id,
        },
    });
    if (!(retentionPolicyAssignment.retentionPolicy.id == retentionPolicy.id)) {
        throw new Error('Assertion failed');
    }
    if (!(retentionPolicyAssignment.assignedTo.id == folder.id)) {
        throw new Error('Assertion failed');
    }
    const retentionPolicyAssignmentById = await exports.client.retentionPolicyAssignments.getRetentionPolicyAssignmentById(retentionPolicyAssignment.id);
    if (!(retentionPolicyAssignmentById.id == retentionPolicyAssignment.id)) {
        throw new Error('Assertion failed');
    }
    const retentionPolicyAssignments = await exports.client.retentionPolicyAssignments.getRetentionPolicyAssignments(retentionPolicy.id);
    if (!(retentionPolicyAssignments.entries.length == 1)) {
        throw new Error('Assertion failed');
    }
    const filesUnderRetention = await exports.client.retentionPolicyAssignments.getFilesUnderRetentionPolicyAssignment(retentionPolicyAssignment.id);
    if (!(filesUnderRetention.entries.length == 1)) {
        throw new Error('Assertion failed');
    }
    await exports.client.retentionPolicyAssignments.deleteRetentionPolicyAssignmentById(retentionPolicyAssignment.id);
    const retentionPolicyAssignmentsAfterDelete = await exports.client.retentionPolicyAssignments.getRetentionPolicyAssignments(retentionPolicy.id);
    if (!(retentionPolicyAssignmentsAfterDelete.entries.length == 0)) {
        throw new Error('Assertion failed');
    }
    await exports.client.retentionPolicies.deleteRetentionPolicyById(retentionPolicy.id);
    await exports.client.files.deleteFileById(file.id);
});
