// box-typescript-sdk-gen/lib-esm/test/shieldInformationBarrierSegments.generated.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const utils_js_1 = require("../internal/utils.js");
const utils_js_2 = require("../internal/utils.js");
const commons_generated_js_1 = require("./commons.generated.js");
const commons_generated_js_2 = require("./commons.generated.js");
const utils_js_3 = require("../internal/utils.js");
test('testShieldInformationBarrierSegments', async function testShieldInformationBarrierSegments() {
    const client = (0, commons_generated_js_1.getDefaultClientWithUserSubject)((0, utils_js_1.getEnvVar)('USER_ID'));
    const enterpriseId = (0, utils_js_1.getEnvVar)('ENTERPRISE_ID');
    const barrier = await (0, commons_generated_js_2.getOrCreateShieldInformationBarrier)(client, enterpriseId);
    const barrierId = barrier.id;
    const segmentName = (0, utils_js_2.getUuid)();
    const segmentDescription = 'barrier segment description';
    const segment = await client.shieldInformationBarrierSegments.createShieldInformationBarrierSegment({
        shieldInformationBarrier: {
            id: barrierId,
            type: 'shield_information_barrier',
        },
        name: segmentName,
        description: segmentDescription,
    });
    if (!(segment.name == segmentName)) {
        throw new Error('Assertion failed');
    }
    const segments = await client.shieldInformationBarrierSegments.getShieldInformationBarrierSegments({
        shieldInformationBarrierId: barrierId,
    });
    if (!(segments.entries.length > 0)) {
        throw new Error('Assertion failed');
    }
    const segmentId = segment.id;
    const segmentFromApi = await client.shieldInformationBarrierSegments.getShieldInformationBarrierSegmentById(segmentId);
    if (!((0, utils_js_3.toString)(segmentFromApi.type) ==
        'shield_information_barrier_segment')) {
        throw new Error('Assertion failed');
    }
    if (!(segmentFromApi.id == segmentId)) {
        throw new Error('Assertion failed');
    }
    if (!(segmentFromApi.name == segmentName)) {
        throw new Error('Assertion failed');
    }
    if (!(segmentFromApi.description == segmentDescription)) {
        throw new Error('Assertion failed');
    }
    if (!(segmentFromApi.shieldInformationBarrier.id == barrierId)) {
        throw new Error('Assertion failed');
    }
    const updatedSegmentDescription = 'updated barrier segment description';
    const updatedSegment = await client.shieldInformationBarrierSegments.updateShieldInformationBarrierSegmentById(segmentId, {
        requestBody: {
            description: updatedSegmentDescription,
        },
    });
    if (!(updatedSegment.description == updatedSegmentDescription)) {
        throw new Error('Assertion failed');
    }
    await client.shieldInformationBarrierSegments.deleteShieldInformationBarrierSegmentById(segmentId);
    await expect(async () => {
        await client.shieldInformationBarrierSegments.getShieldInformationBarrierSegmentById(segmentId);
    }).rejects.toThrow();
});
