// box-typescript-sdk-gen/lib-esm/test/shieldInformationBarrierSegmentMembers.generated.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const userBase_generated_js_1 = require("../schemas/userBase.generated.js");
const utils_js_1 = require("../internal/utils.js");
const utils_js_2 = require("../internal/utils.js");
const commons_generated_js_1 = require("./commons.generated.js");
const commons_generated_js_2 = require("./commons.generated.js");
test('testShieldInformationBarrierSegmentMembers', async function testShieldInformationBarrierSegmentMembers() {
    const client = (0, commons_generated_js_1.getDefaultClientWithUserSubject)((0, utils_js_1.getEnvVar)('USER_ID'));
    const enterpriseId = (0, utils_js_1.getEnvVar)('ENTERPRISE_ID');
    const barrier = await (0, commons_generated_js_2.getOrCreateShieldInformationBarrier)(client, enterpriseId);
    const barrierId = barrier.id;
    const segmentName = (0, utils_js_2.getUuid)();
    const segment = await client.shieldInformationBarrierSegments.createShieldInformationBarrierSegment({
        shieldInformationBarrier: {
            id: barrierId,
            type: 'shield_information_barrier',
        },
        name: segmentName,
    });
    if (!(segment.name == segmentName)) {
        throw new Error('Assertion failed');
    }
    const segmentMember = await client.shieldInformationBarrierSegmentMembers.createShieldInformationBarrierSegmentMember({
        shieldInformationBarrierSegment: {
            id: segment.id,
            type: 'shield_information_barrier_segment',
        },
        user: new userBase_generated_js_1.UserBase({ id: (0, utils_js_1.getEnvVar)('USER_ID') }),
    });
    if (!(segmentMember.user.id == (0, utils_js_1.getEnvVar)('USER_ID'))) {
        throw new Error('Assertion failed');
    }
    if (!(segmentMember.shieldInformationBarrierSegment.id == segment.id)) {
        throw new Error('Assertion failed');
    }
    const segmentMembers = await client.shieldInformationBarrierSegmentMembers.getShieldInformationBarrierSegmentMembers({
        shieldInformationBarrierSegmentId: segment.id,
    });
    if (!(segmentMembers.entries.length > 0)) {
        throw new Error('Assertion failed');
    }
    const segmentMemberGet = await client.shieldInformationBarrierSegmentMembers.getShieldInformationBarrierSegmentMemberById(segmentMember.id);
    if (!(segmentMemberGet.id == segmentMember.id)) {
        throw new Error('Assertion failed');
    }
    await client.shieldInformationBarrierSegmentMembers.deleteShieldInformationBarrierSegmentMemberById(segmentMember.id);
    await expect(async () => {
        await client.shieldInformationBarrierSegmentMembers.getShieldInformationBarrierSegmentMemberById(segmentMember.id);
    }).rejects.toThrow();
    await client.shieldInformationBarrierSegments.deleteShieldInformationBarrierSegmentById(segment.id);
});
