// box-typescript-sdk-gen/lib-esm/test/shieldInformationBarrierSegmentRestrictions.generated.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const utils_js_1 = require("../internal/utils.js");
const utils_js_2 = require("../internal/utils.js");
const commons_generated_js_1 = require("./commons.generated.js");
const commons_generated_js_2 = require("./commons.generated.js");
test('testShieldInformationBarrierSegmentRestrictions', async function testShieldInformationBarrierSegmentRestrictions() {
    const client = (0, commons_generated_js_1.getDefaultClientWithUserSubject)((0, utils_js_1.getEnvVar)('USER_ID'));
    const enterpriseId = (0, utils_js_1.getEnvVar)('ENTERPRISE_ID');
    const barrier = await (0, commons_generated_js_2.getOrCreateShieldInformationBarrier)(client, enterpriseId);
    const barrierId = barrier.id;
    const segment = await client.shieldInformationBarrierSegments.createShieldInformationBarrierSegment({
        shieldInformationBarrier: {
            id: barrierId,
            type: 'shield_information_barrier',
        },
        name: (0, utils_js_2.getUuid)(),
        description: 'barrier segment description',
    });
    const segmentId = segment.id;
    const segmentToRestrict = await client.shieldInformationBarrierSegments.createShieldInformationBarrierSegment({
        shieldInformationBarrier: {
            id: barrierId,
            type: 'shield_information_barrier',
        },
        name: (0, utils_js_2.getUuid)(),
        description: 'barrier segment description',
    });
    const segmentToRestrictId = segmentToRestrict.id;
    const segmentRestriction = await client.shieldInformationBarrierSegmentRestrictions.createShieldInformationBarrierSegmentRestriction({
        restrictedSegment: {
            id: segmentToRestrictId,
            type: 'shield_information_barrier_segment',
        },
        shieldInformationBarrierSegment: {
            id: segmentId,
            type: 'shield_information_barrier_segment',
        },
        type: 'shield_information_barrier_segment_restriction',
    });
    const segmentRestrictionId = segmentRestriction.id;
    if (!(segmentRestriction.shieldInformationBarrierSegment.id == segmentId)) {
        throw new Error('Assertion failed');
    }
    const segmentRestrictions = await client.shieldInformationBarrierSegmentRestrictions.getShieldInformationBarrierSegmentRestrictions({
        shieldInformationBarrierSegmentId: segmentId,
    });
    if (!(segmentRestrictions.entries.length > 0)) {
        throw new Error('Assertion failed');
    }
    const segmentRestrictionFromApi = await client.shieldInformationBarrierSegmentRestrictions.getShieldInformationBarrierSegmentRestrictionById(segmentRestrictionId);
    if (!(segmentRestrictionFromApi.id == segmentRestrictionId)) {
        throw new Error('Assertion failed');
    }
    if (!(segmentRestrictionFromApi.shieldInformationBarrierSegment.id == segmentId)) {
        throw new Error('Assertion failed');
    }
    if (!(segmentRestrictionFromApi.restrictedSegment.id == segmentToRestrictId)) {
        throw new Error('Assertion failed');
    }
    if (!(segmentRestrictionFromApi.shieldInformationBarrier.id == barrierId)) {
        throw new Error('Assertion failed');
    }
    await client.shieldInformationBarrierSegmentRestrictions.deleteShieldInformationBarrierSegmentRestrictionById(segmentRestrictionId);
    await expect(async () => {
        await client.shieldInformationBarrierSegmentRestrictions.getShieldInformationBarrierSegmentRestrictionById(segmentRestrictionId);
    }).rejects.toThrow();
    await client.shieldInformationBarrierSegments.deleteShieldInformationBarrierSegmentById(segmentId);
    await client.shieldInformationBarrierSegments.deleteShieldInformationBarrierSegmentById(segmentToRestrictId);
});
