// box-typescript-sdk-gen/lib-esm/test/tasks.generated.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.client = void 0;
const utils_js_1 = require("../internal/utils.js");
const utils_js_2 = require("../internal/utils.js");
const utils_js_3 = require("../internal/utils.js");
const utils_js_4 = require("../internal/utils.js");
const commons_generated_js_1 = require("./commons.generated.js");
exports.client = (0, commons_generated_js_1.getDefaultClient)();
test('testCreateUpdateGetDeleteTask', async function testCreateUpdateGetDeleteTask() {
    const files = await exports.client.uploads.uploadFile({
        attributes: {
            name: (0, utils_js_1.getUuid)(),
            parent: { id: '0' },
        },
        file: (0, utils_js_2.generateByteStream)(10),
    });
    const file = files.entries[0];
    const dateTime = (0, utils_js_3.dateTimeFromString)('2035-01-01T00:00:00Z');
    const task = await exports.client.tasks.createTask({
        item: {
            type: 'file',
            id: file.id,
        },
        message: 'test message',
        dueAt: dateTime,
        action: 'review',
        completionRule: 'all_assignees',
    });
    if (!(task.message == 'test message')) {
        throw new Error('Assertion failed');
    }
    if (!(task.item.id == file.id)) {
        throw new Error('Assertion failed');
    }
    if (!((0, utils_js_4.dateTimeToString)(task.dueAt) == (0, utils_js_4.dateTimeToString)(dateTime))) {
        throw new Error('Assertion failed');
    }
    const taskById = await exports.client.tasks.getTaskById(task.id);
    if (!(taskById.id == task.id)) {
        throw new Error('Assertion failed');
    }
    const taskOnFile = await exports.client.tasks.getFileTasks(file.id);
    if (!(taskOnFile.totalCount == 1)) {
        throw new Error('Assertion failed');
    }
    const updatedTask = await exports.client.tasks.updateTaskById(task.id, {
        requestBody: {
            message: 'updated message',
        },
    });
    if (!(updatedTask.message == 'updated message')) {
        throw new Error('Assertion failed');
    }
    await exports.client.tasks.deleteTaskById(task.id);
    await exports.client.files.deleteFileById(file.id);
});
