// box-typescript-sdk-gen/lib-esm/test/transfer.generated.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.client = void 0;
const utils_js_1 = require("../internal/utils.js");
const commons_generated_js_1 = require("./commons.generated.js");
exports.client = (0, commons_generated_js_1.getDefaultClient)();
test('testTransferUserContent', async function testTransferUserContent() {
    const sourceUserName = (0, utils_js_1.getUuid)();
    const sourceUser = await exports.client.users.createUser({
        name: sourceUserName,
        isPlatformAccessOnly: true,
    });
    const targetUser = await exports.client.users.getUserMe();
    const transferredFolder = await exports.client.transfer.transferOwnedFolder(sourceUser.id, {
        ownedBy: {
            id: targetUser.id,
        },
    }, {
        queryParams: { notify: false },
    });
    if (!(transferredFolder.ownedBy.id == targetUser.id)) {
        throw new Error('Assertion failed');
    }
    await exports.client.folders.deleteFolderById(transferredFolder.id, {
        queryParams: { recursive: true },
    });
    await exports.client.users.deleteUserById(sourceUser.id, {
        queryParams: {
            notify: false,
            force: true,
        },
    });
});
