// box-typescript-sdk-gen/lib-esm/test/workflows.generated.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.client = void 0;
const utils_js_1 = require("../internal/utils.js");
const utils_js_2 = require("../internal/utils.js");
const utils_js_3 = require("../internal/utils.js");
const commons_generated_js_1 = require("./commons.generated.js");
const commons_generated_js_2 = require("./commons.generated.js");
const utils_js_4 = require("../internal/utils.js");
exports.client = (0, commons_generated_js_1.getDefaultClient)();
test('testWorkflows', async function testWorkflows() {
    const adminClient = (0, commons_generated_js_2.getDefaultClientWithUserSubject)((0, utils_js_2.getEnvVar)('USER_ID'));
    const workflowFolderId = (0, utils_js_2.getEnvVar)('WORKFLOW_FOLDER_ID');
    const uploadedFiles = await adminClient.uploads.uploadFile({
        attributes: {
            name: (0, utils_js_1.getUuid)(),
            parent: {
                id: workflowFolderId,
            },
        },
        file: (0, utils_js_3.generateByteStream)(1024 * 1024),
    });
    const file = uploadedFiles.entries[0];
    const workflowFileId = file.id;
    const workflows = await adminClient.workflows.getWorkflows({
        folderId: workflowFolderId,
    });
    if (!(workflows.entries.length == 1)) {
        throw new Error('Assertion failed');
    }
    const workflowToRun = workflows.entries[0];
    if (!((0, utils_js_4.toString)(workflowToRun.type) == 'workflow')) {
        throw new Error('Assertion failed');
    }
    if (!(workflowToRun.isEnabled == true)) {
        throw new Error('Assertion failed');
    }
    if (!((0, utils_js_4.toString)(workflowToRun.flows[0].type) == 'flow')) {
        throw new Error('Assertion failed');
    }
    if (!((0, utils_js_4.toString)(workflowToRun.flows[0].trigger.type) == 'trigger')) {
        throw new Error('Assertion failed');
    }
    if (!((0, utils_js_4.toString)(workflowToRun.flows[0].trigger.triggerType) ==
        'WORKFLOW_MANUAL_START')) {
        throw new Error('Assertion failed');
    }
    if (!((0, utils_js_4.toString)(workflowToRun.flows[0].outcomes[0].actionType) ==
        'delete_file')) {
        throw new Error('Assertion failed');
    }
    if (!((0, utils_js_4.toString)(workflowToRun.flows[0].outcomes[0].type) ==
        'outcome')) {
        throw new Error('Assertion failed');
    }
    await adminClient.workflows.startWorkflow(workflowToRun.id, {
        type: 'workflow_parameters',
        flow: {
            type: 'flow',
            id: workflowToRun.flows[0].id,
        },
        files: [
            {
                type: 'file',
                id: workflowFileId,
            },
        ],
        folder: {
            type: 'folder',
            id: workflowFolderId,
        },
    });
});
